upup/pkg/fi/cloudup/scalewaytasks/loadbalancer.go

```diff
@@ -0,0 +1,233 @@
+var _ fi.HasAddress = &LoadBalancer{}
+
+func (l *LoadBalancer) CompareWithID() *string {
+	return l.LoadBalancerId
+}
+
+func (l *LoadBalancer) IsForAPIServer() bool {
+	return l.ForAPIServer
+}
+
+func (l *LoadBalancer) Find(context *fi.CloudupContext) (*LoadBalancer, error) {
+	if fi.ValueOf(l.LoadBalancerId) == "" {
+		return nil, nil
+	}
+
+	cloud := context.T.Cloud.(scaleway.ScwCloud)
+	lbService := cloud.LBService()
+
+	loadBalancer, err := lbService.GetLB(&lb.GetLBRequest{
+		Region: scw.Region(cloud.Region()),
+		LBID:   fi.ValueOf(l.LoadBalancerId),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error getting load-balancer %s: %s", fi.ValueOf(l.LoadBalancerId), err)
+	}
+
+	lbIP := loadBalancer.IP[0].IPAddress
+	if len(loadBalancer.IP) > 1 {
+		klog.V(4).Infof("multiple IPs found for load-balancer, using %s", lbIP)
+	}
+
+	return &LoadBalancer{
+		Name:                &loadBalancer.Name,
+		LoadBalancerId:      &loadBalancer.ID,
+		LoadBalancerAddress: &lbIP,
+		Tags:                loadBalancer.Tags,
+		Lifecycle:           l.Lifecycle,
+		ForAPIServer:        l.ForAPIServer,
+	}, nil
+}
+
+func (l *LoadBalancer) FindAddresses(context *fi.CloudupContext) ([]string, error) {
+	cloud := context.T.Cloud.(scaleway.ScwCloud)
+	lbService := cloud.LBService()
+
+	if l.LoadBalancerId == nil {
+		return nil, nil
+	}
+
+	loadBalancer, err := lbService.GetLB(&lb.GetLBRequest{
+		Region: scw.Region(cloud.Region()),
+		LBID:   fi.ValueOf(l.LoadBalancerId),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	addresses := []string(nil)
+	for _, address := range loadBalancer.IP {
+		addresses = append(addresses, address.IPAddress)
+	}
+
+	return addresses, nil
+}
+
+func (l *LoadBalancer) Run(context *fi.CloudupContext) error {
+	return fi.CloudupDefaultDeltaRunMethod(l, context)
+}
+
+func (_ *LoadBalancer) CheckChanges(actual, expected, changes *LoadBalancer) error {
+	if actual != nil {
+		if changes.Name != nil {
+			return fi.CannotChangeField("Name")
+		}
+		if changes.LoadBalancerId != nil {
+			return fi.CannotChangeField("ID")
+		}
+		if changes.Region != nil {
+			return fi.CannotChangeField("Region")
+		}
+	} else {
+		if expected.Name == nil {
+			return fi.RequiredField("Name")
+		}
+		if expected.Region == nil {
+			return fi.RequiredField("Region")
+		}
+	}
+	return nil
+}
+
+func (l *LoadBalancer) RenderScw(t *scaleway.ScwAPITarget, actual, expected, changes *LoadBalancer) error {
+	lbService := t.Cloud.LBService()
+
+	// We check if the load-balancer already exists
+	lbs, err := lbService.ListLBs(&lb.ListLBsRequest{
+		Region: scw.Region(fi.ValueOf(expected.Region)),
+		Name:   expected.Name,
+	}, scw.WithAllPages())
+	if err != nil {
+		return fmt.Errorf("error listing existing load-balancers: %w", err)
+	}
+
+	if lbs.TotalCount > 0 {
+		loadBalancer := lbs.LBs[0]
+		lbIP := loadBalancer.IP[0].IPAddress
+		if len(loadBalancer.IP) > 1 {
+			klog.V(4).Infof("multiple IPs found for load-balancer, using %s", lbIP)
+		}
```

Comment:
What if the load balancer exists, but there was an error creating its backend or frontend?

Reply:
I reworked this part a bit, but it's still a work in progress so I'll turn the PR back into draft.